### ISSUE TITLE
Multithreaded write cause connection pool saturated.

### ISSUE THREAD

**Issue by channguyen** [requery/requery#679]
Hello @npurushe 

I'm currently using Requerry with Rx on HikariCP pool. From what I see, whenever I run on update/insert query on `Schedulers.io()` or any kind of threadpool (except `Schedulers.single()`), eventually the pool will get saturated, if the number of concurrent request > core pool size. I just want to understand is this by design or I might misconfigure somewhere? Since it will defeat the purpose of Rx where I can run multi query on different thread. This feels like all calls to db must be serial.

**Comment by channguyen** [comment]
When I ran multiple updates on different thread. It's deadlock in `EntityUpdateOperation`. I'm seeing connection return from:

```
@Override
    public synchronized Connection getConnection() throws SQLException {
      Connection connection = null;
      Transaction transaction = transactionProvider.get();
      // if the transaction holds a connection use that
      if (transaction != null && transaction.active()) {
        if (transaction instanceof ConnectionProvider) {
          ConnectionProvider connectionProvider = (ConnectionProvider) transaction;
          connection = connectionProvider.getConnection();
        }
      }
      if (connection == null) {
        connection = connectionProvider.getConnection();
        if (statementCache != null) {
          connection = new StatementCachingConnection(statementCache, connection);
        }
      }
      // lazily create things that depend on a connection
      if (platform == null) {
        platform = new PlatformDelegate(connection);
      }
      if (mapping == null) {
        mapping = new GenericMapping(platform);
      }
      return connection;
    }
```
But this block inside never execute:
```
try (Connection connection = configuration.getConnection()) {
          StatementListener listener = configuration.getStatementListener();
          try (PreparedStatement statement = prepare(sql, connection)) {
            bindParameters(statement);
            listener.beforeExecuteUpdate(statement, sql, null);
            result = statement.executeUpdate();
            listener.afterExecuteUpdate(statement, result);
            readGeneratedKeys(0, statement);
          }
        } catch (SQLException e) {
          throw new StatementExecutionException(e, sql);
        }
```
Do you have any idea what would cause this?

**Comment by npurushe** [comment]
Do you have any stacktrace with method/line numbers on the deadlocked threads? thanks

**Comment by channguyen** [comment]
@npurushe 
The stacktrace is very similar to this issue
https://github.com/requery/requery/issues/550

It also caused Posgres deadlock when we run this in prod.
One thing I noticed though, if the number of concurrent request < max pool size. Everything works perfectly fine. A workaround that I did was to put all DB calls to a fixed size threadpool with max size = max_pool_size - 1. 

When I run through my debugger, I think it's deadlock in `EntityUpdateOperation` because `configuration.getConnection()` is called. The snippet I posted above is the easiest to reproduce this issue. One simple `User` table with Hikari connection will do.
